src/equisolve/numpy/preprocessing/_base.py

```diff
@@ -39,6 +48,7 @@ def __init__(
         column_wise: bool = False,
         rtol: float = 0.0,
         atol: float = 1e-12,
+        gradient_compliant: bool = True,
```

Comment:
Why do we need this flag? I don't see a use case for `gradient_compliant=False` at all, so it feels like this is only adding needless complexity =/

Reply:
It goes back to the discussion of https://github.com/lab-cosmo/equistore/issues/118
I am not sure, I would like to try it out. Its not much added complexity in terms of code and it just makes the user aware of a complexity that he should be anyway aware of. So in the worst case this is not used and it only makes the user aware of what is happening here.

Reply:
I would apply [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) here. If we don't have a use case for it, do not add it, even if it is a simple change, because a lot of unnecessary little changes can lead to death by a thousand cuts. 

If you want to play with it, this change does not have to be in the main branch, it can stay in a personal branch until we find a usecase for it.

> makes the user aware of what is happening here.

I would rather do this with documentation on the `with_mean` parameter, saying there that only the values are modified.

Reply:
Fair enough, I thought about it and I cannot think of any transformer benefiting of this option, and if it turns out to be useful somewhere else, we can *then* think about.